### PR TITLE
Studio function to create an example Policyfile

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Adding a new studio functions `push_example_policyfile` and `converge_policyfile_chef_client` that create a CCR with an example policyfile. This is useful when testing policyfiles with Infra nodes or the event feed. The `converge_policyfile_chef_client` function first calls the `push_example_policyfile` function to create the example policyfile and upload it to Chef-Server.

### :athletic_shoe: How to Build and Test the Change
1. From the Automate hab studio run `start_all_services`
1. To start the Automate chef infra server run `start_chef_server`
1. To create a CCR with an example policyfile run `converge_policyfile_chef_client`. This function first creates the example policyfile by calling the `push_example_policyfile` function. 
1. Goto the event feed in the UI to see the new events https://a2-dev.test/dashboards/event-feed (image below)
1. Goto the infrastructure tab in the UI to see the policyfile CCR node named "test-policyfile" https://a2-dev.test/infrastructure/client-runs (image below)

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Event Feed
policyfile

Infrastructure
policyfile_node

### CODE REVIEW

**Review by newblaque — APPROVED** [review]

File: .studio/chef-server-collection [hunk 1]
```diff
@@ -70,6 +71,19 @@ ssl_verify_mode        :verify_none
...
+log_location           STDOUT
+chef_server_url        'https://${chef_server_hostname}/organizations/${chef_server_test_org_name}'
+node_name              '${chef_server_test_node_name}'
+ssl_verify_mode        :verify_none
+client_key             '${chef_server_test_admin_key_path_path}'
+validation_client_name 'test-validator'
+validation_key         '${chef_server_test_validator_path}'
+policy_group           'test_policy_group'
+policy_name            'examplecb'
+use_policyfile         true
```
Comment (lancewf) [inline .studio/chef-server-collection:86]: Adding a new client.rb file to add the policyfile information.

File: .studio/chef-server-collection [hunk 2]
```diff
@@ -134,6 +148,36 @@ EOH
   berks upload -b ${chef_server_berksfile_path} -c ${chef_server_berksfile_config_path}
 }
 
+document "push_policy_update" <<DOC
+  Pushes a policyfile update to chef server with a cookbook. This also creates a policyfile update 
+  in the event feed. 
+DOC
```
Comment (lancewf) [inline .studio/chef-server-collection:154]: The function creates the policyfile and pushes it to chef server

File: .studio/chef-server-collection [hunk 3]
```diff
@@ -134,6 +148,36 @@ EOH
...
+  CHEF_LICENSE=accept-no-persist chef install
+
+  CHEF_LICENSE=accept-no-persist chef push test_policy_group -c ${chef_server_policyfile_rb_path}
+
+  popd &> /dev/null || exit
+}
+
+document "converge_policyfile_chef_client" <<DOC
+  Converges a test chef-client with a policyfile
+DOC
```
Comment (lancewf) [inline .studio/chef-server-collection:174]: This function creates a CCR with the policyfile created in the above function.

File: dev-docs/adding-data/adding_test_data.md [hunk 4]
```diff
@@ -88,7 +88,7 @@ from hab studio:
 
 ensure you have `jq` installed
 
-there are three options...
+there are four options...
```
Comment (lancewf) [inline dev-docs/adding-data/adding_test_data.md:91]: I noticed that there were four options now :shrug:

**Comment by msorens** [comment]
You mention `push_example_policyfile` introduced in your preamble but do not show how that comes into play to exercise it.

**Comment by lancewf** [comment]
@msorens Thank you for pointing that out. I will update the exercise to explain how that is used.

File: .studio/chef-server-collection (hunk 1 above)
Comment (danielsdeleo) [reply .studio/chef-server-collection:86]: How did this change the behavior? `policy_group` should come from the CLI argument to `chef push` and `policy_name` should be in the `Policyfile.rb` file. IIRC `use_policyfile` should only change the behavior of the `chef generate` command, but you can force it with `-P` (and policyfile is the default now).

Comment (lancewf) [reply .studio/chef-server-collection:86]: My guess was that this change added the policyfile to the local run of chef-client on line 178. The rest of the references to this file could be replace by the client.rb file above. 
`CHEF_LICENSE=accept-no-persist chef-client -c ${chef_server_policyfile_rb_path}`

Comment (danielsdeleo) [reply .studio/chef-server-collection:86]: Ah, I see it now. We generally suggest to users to set these in the first boot json so I forgot they are referenced in the policyfile-or-not logic in Chef Client. For the use case here it's fine to use the config file though. 👍

**Review by danielsdeleo — APPROVED** [review]

**Pull request merged**